{
  "comments": [
    {
      "key": {
        "uuid": "5f7c97a3_e5adb032",
        "filename": "openstack_requirements/cmds/build_lower_constraints.py",
        "patchSetId": 1
      },
      "lineNbr": 30,
      "author": {
        "id": 13404
      },
      "writtenOn": "2018-07-02T08:55:43Z",
      "side": 1,
      "message": "If `f.read()` fails `body` will not be defined, and the exception can confuse. I would include the `return` inside the `with`, and replace the current one with a `raise`\n\nEdit: as was beautifully done in the next function.",
      "revId": "a396242c4caf339cabc0272c6835da390f57d351",
      "serverId": "4a232e18-c5a9-48ee-94c0-e04e7cca6543",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "5f7c97a3_fe6f7a90",
        "filename": "openstack_requirements/cmds/build_lower_constraints.py",
        "patchSetId": 1
      },
      "lineNbr": 30,
      "author": {
        "id": 2472
      },
      "writtenOn": "2018-07-02T14:46:57Z",
      "side": 1,
      "message": "If read() fails it\u0027s going to throw an exception and the return statement won\u0027t be executed.",
      "parentUuid": "5f7c97a3_e5adb032",
      "revId": "a396242c4caf339cabc0272c6835da390f57d351",
      "serverId": "4a232e18-c5a9-48ee-94c0-e04e7cca6543",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "5f7c97a3_7e028a74",
        "filename": "openstack_requirements/cmds/build_lower_constraints.py",
        "patchSetId": 1
      },
      "lineNbr": 30,
      "author": {
        "id": 13404
      },
      "writtenOn": "2018-07-02T14:56:56Z",
      "side": 1,
      "message": "Oh right!",
      "parentUuid": "5f7c97a3_fe6f7a90",
      "revId": "a396242c4caf339cabc0272c6835da390f57d351",
      "serverId": "4a232e18-c5a9-48ee-94c0-e04e7cca6543",
      "unresolved": false
    }
  ]
}